data/abi/abi_encode.go

```diff
@@ -25,6 +25,25 @@ import (
 	"strings"
 )
 
+// bigIntToBytes casts non-negative big integer to byte slice with specific byte length
+// NOTE: THIS IS A WORKAROUND FOR `fillBytes` METHOD BEFORE GOLANG 1.15+
+//       SHOULD BE REMOVED AFTER WE MOVE TO HIGHER VERSION
+func bigIntToBytes(x *big.Int, byteLen uint) ([]byte, error) {
+	if x.Cmp(big.NewInt(0)) < 0 {
+		return nil, fmt.Errorf("ABI: big Int To Bytes error: should pass in non-negative integer")
+	}
+	if uint(x.BitLen()) > byteLen*8 {
+		return nil, fmt.Errorf("ABI: big Int To Bytes error: integer byte length > given byte length")
+	}
+
+	buffer := make([]byte, byteLen)
+	intBytes := x.Bytes()
+	for i := 0; i < len(intBytes); i++ {
+		buffer[len(buffer)-1-i] = intBytes[len(intBytes)-1-i]
+	}
```

Comment:
Could this have been `copy(buffer, x.Bytes())`?

Reply:
yes, please see Tsachi's comment above.